### PR TITLE
Testdriver API to switch to a window

### PR DESCRIPTION
**Pull request by kereliuk**
This is a PR that adds a testdriver APi to switch to a window given a handle. There needs to be a way to get the required handle name so this feature depends on #10749 and #10747 for now.

(It also needs some docs and an infra test)

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/10752)

### CODE REVIEW

**Comment by kereliuk** [comment]
An example and requested use case for this is for testing lifecycle APIs with page states.

For now I am leaving the infra test for a follow-up CL since I will want other commands that get me the window handles for that :)

**Comment by kereliuk** [comment]
@gsnedders PTAL :)

File: resources/testdriver.js [hunk 1]
```diff
@@ -143,6 +157,20 @@
...
+         *
+         * This matches the behaviour of the {@link
+         * https://w3c.github.io/webdriver/#switch-to-window|WebDriver
+         * Switch to Window command}.
+         *
+         * @returns {Promise} fulfilled window is switched, or rejected in
+         *                    the cases the WebDriver command errors
+         */
+        switch_to_window: function(window_handle) {
+            return window.test_driver_internal.switch_to_window(window_handle);
```
Comment (gsnedders) [inline resources/testdriver.js]: This is just calling itself; should probably be a promise that rejects like the above

File: tools/wptrunner/wptrunner/executors/executorselenium.py [hunk 2]
```diff
@@ -142,6 +143,13 @@ def setup(self):
     def send_keys(self, element, keys):
         return element.send_keys(keys)
 
+class SeleniumSwitchToWindowProtocolPart(SwitchToWindowProtocolPart):
+    def setup(self):
+        self.webdriver = self.parent.webdriver
+
+    def send_keys(self, element, keys):
```
Comment (gsnedders) [inline tools/wptrunner/wptrunner/executors/executorselenium.py]: This is the wrong protocol

**Review by gsnedders — CHANGES_REQUESTED** [review]
Couple of small comments, and needs some in-tree implementation (strongly) preferably. @jgraham might have more comments?

**Comment by foolip** [comment]
This is tracked by https://github.com/web-platform-tests/wpt/issues/10666, closing.

**Pull request closed without merging**